src/GafferScene/DeletePoints.cpp

```diff
@@ -125,23 +242,104 @@ IECore::ConstObjectPtr DeletePoints::computeProcessedObject( const ScenePath &pa
 		return inputObject;
 	}
 
-	std::string deletePrimVarName = pointsPlug()->getValue();
+	SelectionMode selectionMode = (SelectionMode)selectionModePlug()->getValue();
 
-	if( deletePrimVarName.empty() )
+	IECoreScene::PrimitiveVariable toDelete;
+
+	if( selectionMode == SelectionMode::VertexPrimitiveVariable )
 	{
-		return inputObject;
+		std::string deletePrimVarName = pointsPlug()->getValue();
+
+		if( deletePrimVarName.empty() )
+		{
+			return inputObject;
+		}
+
+		PrimitiveVariableMap::const_iterator it = points->variables.find( deletePrimVarName );
+		if( it == points->variables.end() )
+		{
+			if( ignoreMissingVariablePlug()->getValue() )
+			{
+				return inputObject;
+			}
+
+			throw InvalidArgumentException( fmt::format( "DeletePoints : No primitive variable \"{}\" found", deletePrimVarName ) );
+		}
+
+		toDelete = it->second;
 	}
 
-	PrimitiveVariableMap::const_iterator it = points->variables.find( deletePrimVarName );
-	if( it == points->variables.end() )
+
+
+	if( selectionMode == SelectionMode::IdListPrimitiveVariable || selectionMode == SelectionMode::IdList )
 	{
-		if( ignoreMissingVariablePlug()->getValue() )
+		IdData idList;
+		ConstInt64VectorDataPtr idListData;
+
+		if( selectionMode == SelectionMode::IdListPrimitiveVariable )
 		{
-			return inputObject;
+			std::string idListVarName = idListVariablePlug()->getValue();
+
+			if( idListVarName.empty() )
+			{
+				return inputObject;
+			}
+
+			idList.initialize( points, idListVarName, /* throwIfMissing = */ true );
+		}
+		else
+		{
+			idListData = idListPlug()->getValue();
+			idList.int64Elements = &idListData->readable();
+		}
+
+
+		IdData ids;
+		ids.initialize( points, idPlug()->getValue() );
+
+		size_t numPoints = points->getNumPoints();
+		size_t numIds = idList.size();
+
+		BoolVectorDataPtr inactiveData = new BoolVectorData();
+		std::vector<bool> &inactive = inactiveData->writable();
+		inactive.resize( numPoints, false );
+
+
+		if( ids.size() )
+		{
+			std::unordered_map< int64_t, unsigned int > idMap;
```

Comment:
`unsigned int -> size_t`?

Reply:
I presume that to deal with the "duplicate id" case, we'd use an `unordered_set` built from `idList`, and then do a lookup into that for each vertex? I _think_ this would be slightly faster - `insertions + lookups` would be the same as before, but the set would be smaller, and we'd only visit each address in `inactive` once (using `push_back`) instead of initialising the whole thing and then poking into random parts of it. Does that make sense?

Reply:
>> unsigned int -> size_t?

This is another place where I'm genuinely unsure what the right approach is to the variable type ... do you think that we should be moving towards a place where all our primitive types are allowed to have more than 2^32 vertices?

Anyway, in this specific instance, it's academic, because your suggestion to use a set instead of a map that stores indices works fine.

Reply:
> do you think that we should be moving towards a place where all our primitive types are allowed to have more than 2^32 vertices?

No - at least not while USD and the renderers we care about use `int` for _mesh_ vertex ids. But PointsPrimitive has never had any such limitation, so we should use `size_t` here.